src/deploy/functions/release.js

```diff
@@ -152,8 +152,47 @@ module.exports = function(context, options, payload) {
   var deleteReleaseNames;
   var existingScheduledFunctions;
 
+  // Collect all the functions that have a retry policy
+  var failurePolicyFunctions = functionsInfo.filter((fn) => {
+    return !!fn.failurePolicy;
+  });
+
+  var failurePolicyFunctionLabels = failurePolicyFunctions.map((fn) => {
+    return helper.getFunctionLabel(_.get(fn, "name"));
+  });
+  var retryMessage =
+    "The following functions will be retried in case of failure: " +
+    clc.bold(failurePolicyFunctionLabels.join(", ")) +
+    ". " +
+    "Retried executions are billed as any other execution, and functions are retried repeatedly until they either successfully execute or the maximum retry period has elapsed, which can be up to 7 days. " +
+    "For safety, you might want to ensure that your functions are idempotent; see https://firebase.google.com/docs/functions/retries to learn more.";
+
+  utils.logLabeledWarning("functions", retryMessage);
+
+  let proceedPrompt = Promise.resolve(true);
+  if (options.nonInteractive && !options.force) {
```

Comment:
Should this check _if_ there are any functions with a retry policy?

Reply:
Ooooh ... I think so!  Want to send a PR?

Reply:
(or I can do it, but you deserve the glory)

Reply:
I appreciate the opportunity but I have a _lot_ to do right now. =]

Reply:
@armordog done and released!
https://github.com/firebase/firebase-tools/releases/tag/v8.8.1

Thank you for catching that!  Saved me a lot of angry emails in the morning :-)

Reply:
That was _fast_!
Thanks so much. =]